apps/sms/services/js/threads.js

```diff
@@ -6,16 +6,24 @@
   var threads = new Map();
   var messageMap = new Map();
 
-  function Thread(thread) {
-    var length = Thread.FIELDS.length;
-    var key;
-
-    for (var i = 0; i < length; i++) {
-      key = Thread.FIELDS[i];
-      this[key] = thread[key];
-    }
+  function extractSupportedFields(
+    { id, body, timestamp, participants, lastMessageType, status, isDraft }
+  ) {
+    return {
```

Comment:
If so, why do we need to set the lastMessageSubject while creating the thread in fromDraft & fromMessage?